tests/lib/rules/no-misleading-character-class.js

```diff
@@ -1334,9 +1340,131 @@ ruleTester.run("no-misleading-character-class", rule, {
                 suggestions: null
             }]
         },
+        {
+            code: String.raw`
+
+            // "[" and "]" escaped as "\x5B" and "\u005D"
+            new RegExp("\x5B \\ufe0f\u005D")
+
+            `,
+            errors: [{
+                column: 29,
+                endColumn: 37,
+                messageId: "combiningClass",
+                suggestions: null
+            }]
+        },
+        {
+            code: String.raw`
+
+            // backslash escaped as "\u{5c}"
+            new RegExp("[ \u{5c}ufe0f]")
+
+            `,
+            errors: [{
+                column: 26,
+                endColumn: 38,
+                messageId: "combiningClass",
+                suggestions: null
+            }]
+        },
+        {
+            code: String.raw`
+
+            // "0" escaped as "\60"
+            new RegExp("[ \\ufe\60f]")
+
+            `,
+            languageOptions: { sourceType: "script" },
+            errors: [{
+                column: 26,
+                endColumn: 36,
+                messageId: "combiningClass",
+                suggestions: null
+            }]
+        },
+        {
+            code: String.raw`
 
+            // "e" escaped as "\e"
+            new RegExp("[ \\uf\e0f]")
+
+            `,
+            errors: [{
+                column: 26,
+                endColumn: 35,
+                messageId: "combiningClass",
+                suggestions: null
+            }]
+        },
+        {
+            code: String.raw`
+
+            // line continuation: backslash + <CR> + <LF>
+            new RegExp('[ \\u<line continuation>fe0f]')
+
+            `.replace("<line continuation>", "\\\r\n"),
+            errors: [{
+                line: 4,
+                column: 26,
+                endLine: 5,
+                endColumn: 5,
+                messageId: "combiningClass",
+                suggestions: null
+            }]
+        },
+        {
+            code: String.raw`
+
+            // just a backslash escaped as "\\"
+            new RegExp(<backtick>[.\\u200D.]<backtick>)
+
+            `.replaceAll("<backtick>", "`"),
+            errors: [{
+                column: 26,
+                endColumn: 35,
+                messageId: "zwj",
+                suggestions: null
+            }]
+        },
+        {
+            code: String.raw`
+
+            // "u" escaped as "\x75"
+            new RegExp(<backtick>[.\\\x75200D.]<backtick>)
+
+            `.replaceAll("<backtick>", "`"),
+            errors: [{
+                column: 26,
+                endColumn: 38,
+                messageId: "zwj",
+                suggestions: null
+            }]
+        },
+
+        /* eslint-disable lines-around-comment -- see https://github.com/eslint/eslint/issues/18081 */
```

Comment:
Well, that's not for the comment inside the `code`, it's [this comment](https://github.com/eslint/eslint/blob/45006ecf81ad9105dba90a9bf80805b763e9e7b7/tests/lib/rules/no-misleading-character-class.js#L1464) that is producing errors now. I though that the comments in the code would make it easier to locate a test in the source code given its title, because the `code` as it's calculated here doesn't match what is printed in the console.